### PR TITLE
fix test_activation by lowering threshold + validate eps for check_numeric_gradient

### PR DESCRIPTION
**Pull request by azai91**
## Description ##
Address problem with https://github.com/apache/incubator-mxnet/issues/12377 by setting threshold my appropriately. Ran test with 10000 random seeds and did not produce error.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] set activation to 1e-5

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: tests/python/mkl/test_mkldnn.py [hunk 1]
```diff
@@ -292,7 +291,7 @@ def check_activation_training(stype):
             in_location = [mx.nd.array(data_tmp).tostype(stype)]
 
             test = mx.symbol.Activation(data, act_type="relu")
-            check_numeric_gradient(test, in_location, numeric_eps=1e-2, rtol=0.16, atol=1e-4)
+            check_numeric_gradient(test, in_location, numeric_eps=1e-5, rtol=0.16, atol=1e-4)
```
[the remark this answers is not in the document]
Comment (azai91) [reply tests/python/mkl/test_mkldnn.py]: there's a significant difference between using 1e-5 vs 1e-6. I commented in https://github.com/apache/incubator-mxnet/issues/12377. in short, you should never use anything less than 1e-5 as the floats do not have enough precision to calculate the difference in the numerator.

**Comment by kalyc** [comment]
Thanks for your contribution @azai91 
Could you update the PR title to be more descriptive?

@mxnet-label-bot[pr-awaiting-review]

File: tests/python/mkl/test_mkldnn.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lebeg) [reply tests/python/mkl/test_mkldnn.py]: Ok, thanks for the explanation!

**Comment by lupesko** [comment]
Flagging for @anirudh2290 @sandeep-krishnamurthy @nswamy for review/merge.

**Pull request merged**